workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -37,29 +36,39 @@ protected override void OnCreateManager()
             );
         }
 
-        public void QueuePlayerCreationRequest(Vector3 spawnPosition = default, byte[] serializedArguments = null)
+        public void ResetPlayerCreationArguments()
         {
-            playerCreationRequested = true;
-            this.spawnPosition = Vector3f.FromUnityVector(spawnPosition);
+            spawnPosition = Vector3.zero;
+            serializedArguments = null;
+        }
+
+        public void SetPlayerCreationArguments(Vector3 spawnPosition, byte[] serializedArguments)
+        {
+            this.spawnPosition = spawnPosition;
             this.serializedArguments = serializedArguments;
         }
 
+        public void RequestPlayerCreation()
+        {
+            playerCreationRequested = true;
+        }
+
         protected override void OnUpdate()
         {
             if (PlayerLifecycleConfig.AutoRequestPlayerCreation)
             {
                 var initEntities = initializationGroup.GetEntityArray();
                 for (var i = 0; i < initEntities.Length; ++i)
                 {
-                    QueuePlayerCreationRequest();
+                    RequestPlayerCreation();
                 }
             }
 
             if (playerCreationRequested)
             {
                 var request = new CreatePlayerRequestType
                 {
-                    Position = spawnPosition
+                    Position = Vector3f.FromUnityVector(spawnPosition)
```

Comment:
I still think we shouldn't mandate you add a position - the client shouldn't be telling the server where it should spawn anyway